src/atomic_store.rs

```diff
@@ -211,6 +212,9 @@ pub struct AtomicStore {
     file_counter: u32,
     last_counter: Option<u32>,
     resources: HashMap<String, Arc<RwLock<VersionSyncHandle>>>,
+    // How long `commit_version` will wait for resource versions before returning an error.
+    // defaults to 100 milliseconds
+    commit_timeout: Duration,
```

Comment:
One thing I don't love about this: it's going to absolutely break stepping through a debugger, if one of the logs is being committed in another thread.

Maybe we can put in something to disable the timeout mechanism by way of an env variable, that we can set when debugging.

Reply:
I added a check on `AtomicStore::open` that will set the timeout to `Duration::MAX` if the `ATOMIC_STORE_NO_TIMEOUT` environment variable is set